sdk/lib/core/uri.dart

```diff
@@ -95,7 +131,11 @@ const String _hexDigits = "0123456789ABCDEF";
 /// print(uri.fragment); // utility-classes
 /// print(uri.hasQuery); // false
 /// print(uri.data); // null
-/// ```
+/// ``` 
+/// 
+/// Please note that this performs the parsing at runtime, which slows
+/// down your application, as opposed to the explicit declaration
+/// using one of the other constructors.
```

Comment:
.... well, actually ... 
It's true that parsing has a cost, but every other constructor also does *some* parsing and normalization. 

If the URI is particularly simple (nothing needs to be escaped or unescaped, simple stuff like `https://example.com/banana/hut.html` which most actual URIs are, it is probably *more* efficient to use `Uri.parse`, because it just scans the string once, decides that it is "simple", and stores it as-is.

The other constructors require multiple strings as input, will check each for whether to escape or not, and will then remember all the individual parts. I think it caches the concatenation if you call `toString`, so it also takes more memory.

TL;DR: Parsing *simple* URIs is probably the most efficient way to create a URI.

In general, users should not worry about performance of `Uri`, they should do what is best for the program logic. If they end up URI being a bottleneck, *then* they can start looking for optimizations. Most programs do many other things than parsing URIs.